src/org/skypro/skyshop/product/SearchEngine.java

```diff
@@ -0,0 +1,72 @@
+package org.skypro.skyshop.product;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.TreeMap;
+
+import org.skypro.skyshop.product.BestResultNotFoundException;
+
+public class SearchEngine {
+    private final List<Searchable> searchableArray;
+
+    public SearchEngine() {
+        searchableArray = new LinkedList<>();
+    }
+
+    public void add(Searchable item) {
+        searchableArray.add(item);
+    }
+
+    public TreeMap<String, Searchable> search(String term) {
```

Comment:
возвращать нужно Map, чтобы использовать преимущество полиморфизма
```suggestion
    public Map<String, Searchable> search(String term) {
```